Release Notes for v26.1-v26.1.0-beta.3 (#22089)
cockroachdb/docs: src/current/_includes/releases/v26.1/v26.1.0-beta.3.md

- Fixed a deadlock that could occur when a statistics creation task panicked. [#160422][#160422]
- Fixed a bug where rolling back a transaction that had just rolled back a savepoint would block other transactions accessing the same rows for five seconds. [#160477][#160477]
- Fixed a bug where CockroachDB could crash when handling decimals with negative scales via the extended PGWire protocol. An error is now returned instead, matching PostgreSQL behavior. [#160561][#160561]
- v26.1.0-beta.1 and v26.1.0-beta.2 versions of
  CockroachDB could encounter a rare process crash when running TTL jobs,
  and this has now been fixed. [#160689][#160689]

<h3 id="v26-1-0-beta-3-performance-improvements">Performance improvements</h3>

- Fixed a performance regression in `pg_catalog.pg_roles` and `pg_catalog.pg_authid` by avoiding privilege lookups for each row in the table. [#160228][#160228]
- Added a new session variable, `distsql_prevent_partitioning_soft_limited_scans`, which, when true, prevents scans with soft limits from being planned as multiple TableReaders by the physical planner. This should decrease the initial setup costs of some fully-distributed query plans. [#160600][#160600]

> - The session variable `distsql_prevent_partitioning_soft_limited_scans` is now enabled by default. This prevents scans with soft limits from being planned as multiple TableReaders, which decreases the initial setup costs of some fully-distributed query plans. [#160600][#160600]

Reviewer: ```suggestion
- Added a new session variable, `distsql_prevent_partitioning_soft_limited_scans`, which defaults to true. When enabled, this prevents scans with soft limits from being planned as multiple TableReaders, which decreases the initial setup costs of some fully-distributed query plans. [#160600][#160600]
```
